app/models/host.rb

```diff
@@ -1346,22 +1346,17 @@ def first_cat_entry(name)
     Classification.first_cat_entry(name, self)
   end
 
-  # TODO: Rename this to scan_queue and rename scan_from_queue to scan to match
-  #   standard from other places.
-  def scan(userid = "system", _options = {})
+  def scan(userid = "system", options = {})
     log_target = "#{self.class.name} name: [#{name}], id: [#{id}]"
-
-    task = MiqTask.create(:name => "SmartState Analysis for '#{name}' ", :userid => userid)
-
     _log.info("Requesting scan of #{log_target}")
-    begin
-      MiqEvent.raise_evm_job_event(self, :type => "scan", :prefix => "request")
-    rescue => err
-      _log.warn("Error raising request scan event for #{log_target}: #{err.message}")
-      return
-    end
+    check_policy_prevent(:request_host_scan, :scan_queue, userid, options)
```

Comment:
The host scan does need to be queued and we already do that with the existing `scan_from_queue` method.  

My question was asking if we could get away without having to call the new `scan_queue` which then queues the actual scan.  Might be too much effort to do this using `check_policy_prevent` but it seemed worth looking into.